controllers/users.js

```diff
@@ -593,6 +595,74 @@ const filterUsers = async (req, res) => {
   }
 };
 
+// one time script function to perform the migration - adding github_user_id field to the document
+const migrate = async (req, res) => {
+  const usersNotFound = [];
+  let countUserFound = 0;
+  let countUserNotFound = 0;
+  try {
+    // Fetch user data from GitHub API for each document in the users collection
+    // divided by 500 because firestore api guarantee that we can process in batch of 500.
+    const usersSnapshot = await firestore.collection("users").get();
+    const totalUsers = usersSnapshot.docs.length;
+    const batchCount = Math.ceil(totalUsers / 500);
+    // Create batch write operations for each batch of documents
+    for (let i = 0; i < batchCount; i++) {
+      const batchDocs = usersSnapshot.docs.slice(i * 500, (i + 1) * 500);
```

Comment:
rather than doing this why not directly fetch just 500 from db itself? something like paginated ones.

Reply:
It will lead to extra reads on our database as in firestore it reads all the documents up from the startAt document id and also this is simpler in terms of code readability